lib/ui/window.dart

```diff
@@ -411,6 +411,39 @@ class Locale {
 ///
 /// There is a single Window instance in the system, which you can
 /// obtain from the [window] property.
+///
+/// ## Insets and Padding
+///
+/// {@animation 300 300 https://flutter.github.io/assets-for-api-docs/assets/widgets/window_padding.mp4}
+///
+/// In this diagram, the black areas represent system UI that the app cannot
+/// draw over. The red area represents view padding that the application may not
+/// be able to detect gestures in and may not want to draw in. The grey area
+/// represents the system keyboard, which can cover over the bottom view
+/// padding when visible.
+///
+/// The [Window.viewInsets] are the physical pixels into which the operating
+/// system may place system UI, such as the keyboard, which would fully obscure
```

Comment:
may place or did place? In your diagram it looks like this only has a value != 0 when the keyboard is actually up and the screen *is* obscured.

Reply:
This was existing documentation.  Maybe better to say "the system reserves for system UI"?